test/test.js

```diff
@@ -1579,13 +1577,18 @@ describe('Page', function() {
       const requests = [];
       page.on('request', request => requests.push(request));
       await page.goto(server.EMPTY_PAGE);
-      expect(requests.length).toBe(1);
-      expect(requests[0].url()).toContain('empty.html');
+      await FrameUtils.attachFrame(page, 'frame1', server.EMPTY_PAGE);
+      expect(requests.length).toBe(2);
+      expect(requests[0].url()).toBe(server.EMPTY_PAGE);
+      expect(requests[0].frame() === page.mainFrame()).toBe(true);
+      expect(requests[0].frame().url()).toBe(server.EMPTY_PAGE);
+      expect(requests[1].url()).toBe(server.EMPTY_PAGE);
+      expect(requests[1].frame() === page.mainFrame()).toBe(false);
```

Comment:
nit: can you please compare with the frames[1] instead?
```js
expect(requests[1].frame() === page.frames()[1]).toBe(true);
```

Reply:
https://github.com/GoogleChrome/puppeteer/pull/1646/commits/47c93c516f5f0c4e7927f5d878272d0f4e0a61b7